src/builtins.c

```diff
@@ -993,10 +1013,10 @@ JL_DLLEXPORT void jl_show(jl_value_t *stream, jl_value_t *v)
 JL_CALLABLE(jl_f_apply_type)
 {
     JL_NARGSV(apply_type, 1);
-    if (!jl_is_datatype(args[0]) && !jl_is_typector(args[0])) {
-        jl_type_error("Type{...} expression", (jl_value_t*)jl_type_type, args[0]);
-    }
-    return jl_apply_type_(args[0], &args[1], nargs-1);
+    if (!jl_is_unionall(args[0]) && args[0] != (jl_value_t*)jl_anytuple_type &&
```

Comment:
In theory, `A == B` does not guarantee that A and B accept the same type applications. For example, `Int where T` equals `Int` but can be applied to one argument. Of course we simplify `Int where T`, but the basic idea remains. However `NTuple == Tuple{Vararg{T} where T}` itself seems to be a bug; I don't think those should be equal.